### PR TITLE
Support callable for GenericOAuthenticator username_key

### PR DESCRIPTION
**Pull request by eslavich**
We have an OAuth provider that returns the username in a nested object.  This PR updates GenericOAuthenticator's username_key traitlet to support a callable that can be used to extract a username from anywhere in the response.

### CODE REVIEW

File: oauthenticator/traitlets.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from traitlets import TraitType
```
Comment (yuvipanda) [inline oauthenticator/traitlets.py]: We depend on JupyterHub, so you might be able to just import https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/traitlets.py#L93

Comment (eslavich) [reply oauthenticator/traitlets.py]: Ah!  I didn't notice that, thanks.

The requirements.txt allows jupyterhub 0.5, which doesn't include Callable -- Callable first shows up in 0.9.0.  Is it okay to stop supporting jupyterhub < 0.9.0?

Comment (yuvipanda) [reply oauthenticator/traitlets.py]: Oh, that's interesting. I *think* we passed that bridge a long time ago...

Regardless, let's leave it as is and pick that up in a separate PR.

**Comment by yuvipanda** [comment]
Thanks, @eslavich!

**Pull request merged**